apps/src/sites/studio/pages/components/editorRedux.js

```diff
@@ -0,0 +1,127 @@
+/* global levelKeyList */
+
+import _ from 'lodash';
+
+function updatePositions(node) {
+  for (var i = 0; i < node.length; i++) {
+    node[i].position = i + 1;
+  }
+}
+
+let newStageId = -1;
+let newLevelId = -1;
+
+export default function reducer(state, action) {
+  let newState = _.cloneDeep(state);
+
+  switch (action.type) {
+    case 'REORDER_LEVEL': {
```

Comment:
consider prefixing these action type strings with something indicating what domain these actions are relevant for. This makes it easier to combine multiple reducers in a larger application without worrying about action type collisions.